### PR TITLE
Added docker development environment

### PR DESCRIPTION
**Pull request by ElijahCaine**
Run the script `dockersetup.sh -b` to build the environment.
Run the script alone to start and enter a dev container.
Run the script with the `-h` or `--help` flag for help.
Add packages to the Dockerfile when you need them installed.

Re-build when necessary.

This should help make development easier. I could be wrong.

@matthewrsj @Kennric Thoughts?

> Spoiler, it's just a bash script that wraps around some docker commands.

### CODE REVIEW

**Comment by matthewrsj** [comment]
I think this is a good idea. I'll have to relearn how to use docker but that should be trivial since you already wrote the setup script. I was able to successfully run the script and generate the docker image on my machine, so at the very least it doesn't hurt anything.

tl;dr +1

**Comment by Kennric** [comment]
I am going to close this and re-PR against develop, resolve conflicts and merge there.
